doc/nvim-treesitter.txt

```diff
@@ -449,228 +449,236 @@ Note: This is highly experimental, and folding can break on some types of
 `TSInclude`
-For includes: `#include` in C, `use` or `extern crate` in Rust, or `require`
-in Lua.
+File or module inclusion keywords: `#include` in C, `use` or `extern crate` in
+Rust.
 
 								*hl-TSKeyword*
 `TSKeyword`
-For keywords that don't fall in previous categories.
+Keywords that don't fit into other categories.
 
 							*hl-TSKeywordFunction*
 `TSKeywordFunction`
-For keywords used to define a fuction.
+Keywords used to define a function: `function` in Lua, `def` and `lambda` in
+Python.
 
 							*hl-TSKeywordOperator*
 `TSKeywordOperator`
-for operators that are English words, e.g. `and`, `as`, `or`.
+Unary and binary operators that are English words: `and`, `or` in Python;
+`sizeof` in C.
 
 							  *hl-TSKeywordReturn*
 `TSKeywordReturn`
-for the `return` and `yield` keywords.
+Keywords like `return` and `yield`.
 
 								  *hl-TSLabel*
 `TSLabel`
-For labels: `label:` in C and `:label:` in Lua.
+GOTO labels: `label:` in C, and `::label::` in Lua.
 
 								 *hl-TSMethod*
 `TSMethod`
-For method calls and definitions.
+Method calls and definitions.
 
 							      *hl-TSNamespace*
 `TSNamespace`
-For identifiers referring to modules and namespaces.
+Identifiers referring to modules and namespaces.
 
 								     *hl-None*
 `TSNone`
-For no highlighting.
+No highlighting.
 
 								 *hl-TSNumber*
 `TSNumber`
-For all numbers
+Any numeric literal that doesn't fit into other categories.
 
 							       *hl-TSOperator*
 `TSOperator`
-For any operator: `+`, but also `->` and `*` in C.
+Binary or unary operators: `+`, and also `->` and `*` in C.
 
 							      *hl-TSParameter*
 `TSParameter`
-For parameters of a function.
+Parameters of a function.
 
 						     *hl-TSParameterReference*
 `TSParameterReference`
-For references to parameters of a function.
+References to parameters of a function.
 
 							       *hl-TSProperty*
 `TSProperty`
 Same as `TSField`.
 
 							 *hl-TSPunctDelimiter*
 `TSPunctDelimiter`
-For delimiters ie: `.`
+Punctuation delimiters: Periods, commas, semicolons, etc.
 
 							   *hl-TSPunctBracket*
 `TSPunctBracket`
-For brackets and parens.
+Brackets, braces, parentheses, etc.
 
 							   *hl-TSPunctSpecial*
 `TSPunctSpecial`
-For special punctutation that does not fall in the catagories before.
+Special punctuation that doesn't fit into the previous categories.
 
 								 *hl-TSRepeat*
 `TSRepeat`
-For keywords related to loops.
+keywords related to loops: `for`, `while`, etc.
 
 								 *hl-TSString*
 `TSString`
-For strings.
+String literals.
 
 							    *hl-TSStringRegex*
 `TSStringRegex`
-For regexes.
+Regular expression literals.
 
 							   *hl-TSStringEscape*
 `TSStringEscape`
-For escape characters within a string.
+Escape characters within a string: `\n`, `\t`, etc.
 
 							  *hl-TSStringSpecial*
 `TSStringSpecial`
-For strings with special meaning that don't fit into the above categories.
+Strings with special meaning that don't fit into the previous categories.
```

Comment:
An example here could be `{` in python's f-strings`f"{var}"`, aka string interpolation.

Reply:
I think string interpolation highlights text like normal variables?

The usages of `string.special` so far are:

<details>
<summary> $ rg 'string\.special' queries/ </summary>

```
queries/elixir/highlights.scm
78:(sigil) @string.special
93:] @string.special

yang/highlights.scm
19:(quoted_range "\"" @string.special)
20:(quoted_range "'" @string.special)
27:(keypath) @string.special
35:  (argument (string) @string.special))

zig/highlights.scm
84:(FormatSequence) @string.special

nix/highlights.scm
28:[ (path) (spath) (uri) ] @string.special

vim/highlights.scm
97:(pattern) @string.special

supercollider/highlights.scm
39:(symbol) @string.special
```

</details>

Reply:
oops, this was punctuation.special

https://github.com/nvim-treesitter/nvim-treesitter/blob/e46b2545b054df0e20aabec3f7bdc2b73f3cfa62/queries/python/highlights.scm#L235-L237